### PR TITLE
Trigger register usage call on image insertion

### LINKED ISSUE #3497: Issue RegisterUsage requests when media is added from the Media3P Tab

### PR DESCRIPTION
**Pull request by winterDroid**
Add registerUsageUrl to attribution object in the MediaResource

Fixes #3497

### CODE REVIEW

File: assets/src/edit-story/app/media/utils/createResource.js [hunk 1]
```diff
@@ -32,7 +32,8 @@ import getTypeFromMime from './getTypeFromMime';
  * Attribution object
  *
  * @typedef {Attribution} Attribution
- * @property {Author} author The author of the media object.
+ * @property {?Author} author The optional author of the media object.
+ * @property {string} registerUsageUrl The url to register the media usage.
```
Comment (diegovar) [inline assets/src/edit-story/app/media/utils/createResource.js]: This should be ?string as Coverr doesn't have registerUsage.

File: assets/src/edit-story/app/media/utils/createResource.js [hunk 2]
```diff
@@ -85,8 +86,7 @@ import getTypeFromMime from './getTypeFromMime';
  * @property {boolean} local Whether the resource has been already uploaded to
  * the server.
  * @property {Object} sizes Object of image sizes.
- * @property {?Attribution} attribution An optional attribution for the
- * resource.
+ * @property {Attribution} attribution An attribution for the resource.
```
Comment (diegovar) [inline assets/src/edit-story/app/media/utils/createResource.js]: This should probably be optional, as Coverr doesn't have per-resource attribution I believe

File: assets/src/edit-story/components/library/panes/media/common/mediaElement.js [hunk 3]
```diff
@@ -156,6 +157,18 @@ const MediaElement = ({
     actions: { handleDrag, handleDrop, setDraggingResource },
   } = useDropTargets();
 
+  const {
+    actions: { registerUsage },
+  } = useMedia3pApi();
+
+  const handleRegisterUsage = useCallback(() => {
+    if (providerType !== ProviderType.LOCAL) {
```
Comment (diegovar) [inline assets/src/edit-story/components/library/panes/media/common/mediaElement.js]: Only for Unsplash.

File: assets/src/edit-story/app/media/utils/createResource.js (hunk 1 above)
Comment (winterDroid) [reply assets/src/edit-story/app/media/utils/createResource.js]: Didn't we agree that we'll still issue a registerUsageUrl for all providers and then just ignore it on the API side? This allows us to keep the client simpler.

Comment (diegovar) [reply assets/src/edit-story/app/media/utils/createResource.js]: I don't think so, I think we agreed we'd use the same approach for registerUsage across all providers, but not that we'd be issuing registerUsage calls from the client for all providers. Whether a provider needs registerUsage can be configured as part of the provider configuration json for example.

File: assets/src/edit-story/app/media/utils/createResource.js (hunk 2 above)
Comment (winterDroid) [reply assets/src/edit-story/app/media/utils/createResource.js]: Done.

File: assets/src/edit-story/components/library/panes/media/common/mediaElement.js (hunk 3 above)
Comment (winterDroid) [reply assets/src/edit-story/components/library/panes/media/common/mediaElement.js]: As discussed offline, made it nullable.

File: assets/src/edit-story/app/media/utils/createResource.js (hunk 1 above)
Comment (winterDroid) [reply assets/src/edit-story/app/media/utils/createResource.js]: As discussed offline, made the url nullable.

**Review by diegovar — APPROVED** [review]

**Pull request merged**